### DISCUSSION TITLE
LTSP ssh auth documentation

### DISCUSSION THREAD

**Discussion by trickkiste** [ltsp/ltsp#196] [Q&A]
I seem to remember, there was a section in the wiki or in a manpage explaining the ltsp authentication mechanism and how exactly /etc/passwd and /etc/group are being merged.

Can't find that anymore.

Can you add such a section please? 

Since the Wiki was turned off now, how would other users contribute to documentation now? Creating patches and PR for https://github.com/ltsp/ltsp.github.io seems a little bit hard compared to the ease of a Wiki. Doesn't this create a one person bottle neck for the effort of documenting your software?

**Comment by alkisg** [comment]
There are [ltsp issues](https://github.com/ltsp/ltsp/issues) that mean "fix this bug in the ltsp code".
There are [community issues](https://github.com/ltsp/community/issues) that mean "let's discuss things that don't involve changes to the ltsp code base".
So for example I moved this issue to the community issues, as it doesn't require ltsp code changes.

It's the same for the wiki.
There's the upstream "wiki", which is the https://github.com/ltsp/ltsp.github.io repository, and only upstream ltsp developers are able to commit or accept pull requests there,
and there's the [community wiki](https://github.com/ltsp/community/wiki) (where you started the wiki page about CephFS/RDB) that everyone can edit.

If, at some point, upstream LTSP developers see that they need a wiki, we might switch to it, but past experience shows that it's not necessary.
Btw, people interested in joining the upstream LTSP developers team are most welcome. :smile:

About pamltsp, there are a few user-side notes in `man ltsp.conf`, and there are the comments inside the code, mainly pwmerge. I don't remember writing any wiki pages about it.

**Comment by alkisg** [comment]
Btw, if LTSP developers go e.g. to https://github.com/ltsp/ltsp.github.io/blob/master/docs/installation.md and click on "edit", they get the same editor that is used for github wikies; there's no real difference in the editing abilities.

The only real difference from using an open wiki is that the "ltsp.github.io maintainers team" must have specific members; which is something we do want for the official LTSP documentation, to maintain a certain quality.

If someone wants to help with the site or documentation, just ping, give a small sample of your work, and we'll give you access.

**Comment by trickkiste** [comment]
Hmm, well, so who are the LTSP core/upstream developers currently? On
github I only see yourself.

I like the project and had been working with the old LTSP version for ...
is it 20 years already?
And as you can judge from my pull requests, I am interested in contributing
to your project in terms of functionality as well as in terms of usability
and clarity of code.

That is the reason I am vocal about things I believe are missing or could
be implemented in a fashion, which would improve overall usability.

I voice my concerns here and there, because I want to support you in making
your product better.

One such concern is documentation and community engagement. I believe the
way you setup these parts of the project currently fall into the category
of premature optimization. There is no need to start a community wiki, if
there is no community yet. There is no need to worry, that too many
unchecked contributions to the official Wiki might lead to bad quality of
documentation, if probably there have not been any such contributions to
the Wiki yet at all.

The same is true for "issues".

I believe, you have implemented a few measures here, which are premature
and ultimately close out the community instead of engaging it. I would want
to advise to reconsider these decisions and postpone any such measure to a
point in time, where you actually start to observe a negative impact of
maximum openness.

it is far less work to moderate and correct a few misguided
Wiki documentation entries, than to write all documentation by yourself.
Isn't that true? And ultimately users will benefit from more contributions.

Anyhow, these are my 5 cents on the issue of documentation. I hope you can
find any wisdom in them.

My best regards,
Markus

>

**Comment by alkisg** [comment]
Since the summer when I opened the github repositories, only me and vagrant joined the team: https://github.com/orgs/ltsp/people
We're the ones that have been active LTSP5 developers in the last ...maybe 5 years,
except for Kim, who maintains the RHEL implementation and is welcome to join whenever he wants to.
And of course anyone else that wants to contribute is welcome.
The rest of the old LTSP team isn't working on LTSP nowadays.

You've been using LTSP for 20 years... how many wiki.ltsp.org pages have you contributed? How much code? Did you ask to join the team?

I'm not pointing any fingers. In open source, people will contribute as much as they want to or as much as they need to. Facing the facts, I decided that I can maintain myself ltsp.github.io (along with anyone else that wants to join an official team), and set up the community wiki for occasional contributors.

No, I'm not able to maintain pages that other persons write. Each author should be able to maintain their own pages. E.g. "I don't have that use case at all" or "I wouldn't write it this way" or "I already said most of those things in that other page" etc. So e.g. "fixing" wiki.ltsp.org would need twice the time of rewriting it from zero.

So Markus if you want to do work, you're most welcome; start writing wiki pages in the community wiki or send code like you already start to, and after just a bit, when both us and you know that you want to keep up doing that, we move them upstream. If you just want to point out how things could be designed in a better way... there are thousands of users that can give advice, but only a handful that roll up their sleeves and join the upstream team and do upstream work for years.

Thank you for the input!
